servicetalk-dns-discovery-netty/src/main/java/io/servicetalk/dns/discovery/netty/DnsServiceDiscovererBuilder.java

```diff
@@ -30,26 +30,29 @@
  */
 public interface DnsServiceDiscovererBuilder {
     /**
-     * The minimum allowed TTL. This will be the minimum poll interval.
-     *
-     * @param minTTLSeconds The minimum amount of time a cache entry will be considered valid (in seconds).
-     * @return {@code this}.
-     */
-    DnsServiceDiscovererBuilder minTTL(int minTTLSeconds);
-
-    /**
-     * The maximum allowed TTL. This will be the maximum poll interval as well as the maximum dns cache value.
+     * Controls min/max TTL values that will affect polling interval and caching.
+     * <p>
+     * The created {@link ServiceDiscoverer} polls DNS server based on TTL value of the resolved records. Min/max values
+     * help to make sure polling stays within reasonable boundaries. The 3rd argument controls if the resolved records
+     * should be cached or not. Cache is helpful in scenarios when multiple concurrent resolutions are possible for the
+     * same address: either an application runs multiple client instances for the same remote address (not recommended)
+     * or clients perform DNS resolutions per new connection instead of background polling.
      *
-     * @param maxTTLSeconds the maximum amount of time a cache entry will be considered valid (in seconds).
+     * @param minSeconds The minimum about of time the result will be considered valid (in seconds), must be greater
+     * than {@code 0}.
+     * @param maxSeconds The maximum about of time the result will be considered valid (in seconds), must be greater
+     * than {@code minSeconds}.
+     * @param cache If {@code true}, DNS responses will be cached locally for the specified time. Any concurrent
+     * resolutions for the same address will hit the cache if it's not expired. Otherwise, all resolutions will generate
+     * a new query for DNS server.
      * @return {@code this}.
      */
-    DnsServiceDiscovererBuilder maxTTL(int maxTTLSeconds);
+    DnsServiceDiscovererBuilder ttl(int minSeconds, int maxSeconds, boolean cache);
```

Comment:
If we start using a new method, consider using `Duration` instead of int seconds

Reply:
Also, should we add a different builder for the caching enable/disable?

Reply:
As we discussed offline, I went with:
```
.ttl(min, max)
.ttl(min, max, minCache, maxCache)
```
Thanks for brainstorming!